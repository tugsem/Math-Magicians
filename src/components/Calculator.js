import React, { useState } from 'react';
import calculate from '../logic/calculate.js';

export default function Calculator() {
   const [result, setResult] = useState({total: '', next: '', operation: ''});
    const handleClick = (e) => {
        setResult(calculate(result, e.target.value));
    }
    return (
            <table className='calculator'>
                <tbody>
                    <tr>
                        <td colSpan={4}><input className='displayBox' type='text' value={result.next || result.total || '0'} disabled/></td>
                    </tr>
                    <tr>
                    <td><input className='button' type='button' value='AC' onClick= {handleClick}/></td>
                    <td><input className='button' type='button' value='+/-' onClick= {handleClick}/> </td>
                    <td><input className='button' type='button' value='%' onClick= {handleClick}/></td> 
                    <td><input className='button orange' type='button' value='รท' onClick= {handleClick}/></td> 
                    </tr>
                    <tr>
                    <td><input className='button' type='button' value={7} onClick= {handleClick}/></td>
                    <td><input className='button' type='button' value={8} onClick= {handleClick}/></td> 
                    <td><input className='button' type='button' value={9} onClick= {handleClick}/></td> 
                    <td><input className='button orange' type='button' value='x' onClick= {handleClick}/></td> 
                    </tr>
                    <tr>
                    <td><input className='button' type='button' value={4} onClick= {handleClick}/></td>
                    <td><input className='button' type='button' value={5} onClick= {handleClick}/></td> 
                    <td><input className='button' type='button' value={6} onClick= {handleClick}/></td> 
                    <td><input className='button orange' type='button' value='-' onClick= {handleClick}/></td> 
                    </tr>
                    <tr>
                    <td><input className='button' type='button' value={1} onClick= {handleClick}/></td>
                    <td><input className='button' type='button' value={2} onClick= {handleClick}/></td> 
                    <td><input className='button' type='button' value={3} onClick= {handleClick}/></td> 
                    <td><input className='button orange' type='button' value='+' onClick= {handleClick}/></td>
                    </tr>
                    <tr>
                    <td colSpan={2}><input className='button' type='button' value={0} onClick= {handleClick}/></td>
                    <td><input className='button' type='button' value='.' onClick= {handleClick}/></td> 
                    <td><input className='button orange' type='button' value='=' id='equal' onClick={handleClick}/></td> 
                    </tr>
                </tbody>
            </table>
    )
}
