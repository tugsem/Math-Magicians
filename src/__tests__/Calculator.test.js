import react from 'react';
import renderer  from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('Renders correctly',()=>{
    const tree = renderer.create(
        <Calculator />).toJSON();
        expect(tree).toMatchSnapshot();
})