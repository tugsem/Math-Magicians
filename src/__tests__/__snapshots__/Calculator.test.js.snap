// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="calContainer"
>
  <h1>
    Let's Do Some Math!
  </h1>
  <table
    className="calculator"
  >
    <tbody>
      <tr>
        <td
          colSpan={4}
        >
          <input
            className="displayBox"
            disabled={true}
            type="text"
            value="0"
          />
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="AC"
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="+/-"
          />
           
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="%"
          />
        </td>
        <td>
          <input
            className="button secondColor"
            onClick={[Function]}
            type="button"
            value="รท"
          />
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={7}
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={8}
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={9}
          />
        </td>
        <td>
          <input
            className="button secondColor"
            onClick={[Function]}
            type="button"
            value="x"
          />
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={4}
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={5}
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={6}
          />
        </td>
        <td>
          <input
            className="button secondColor"
            onClick={[Function]}
            type="button"
            value="-"
          />
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={1}
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={2}
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={3}
          />
        </td>
        <td>
          <input
            className="button secondColor"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </td>
      </tr>
      <tr>
        <td
          colSpan={2}
        >
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value={0}
          />
        </td>
        <td>
          <input
            className="button"
            onClick={[Function]}
            type="button"
            value="."
          />
        </td>
        <td>
          <input
            className="button secondColor"
            id="equal"
            onClick={[Function]}
            type="button"
            value="="
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
